api/rest/client/lbclient.go

```diff
@@ -44,20 +45,23 @@ func (r *RoundRobin) SetClients(cl []Client) {
 	r.length = uint32(len(cl))
 }
 
-// Failover is a load balancing strategy that would try the local cluster peer first.
-// If the local call fail it would try other client in a round robin fashion.
+// Failover is a load balancing strategy that would try the local cluster peer
+// first. If the local call fails it would try other clients for that call in a
+// round robin fashion.
 type Failover struct {
 	clients []Client
+	length  int
 }
 
 // Next returns the next client to be used.
 func (f *Failover) Next(count int) Client {
-	return f.clients[count]
+	return f.clients[count%f.length]
```

Comment:
why can't you `count%len(f.clients)` and save a struct field?

Reply:
it save us from calling len() several times.

Reply:
https://blog.golang.org/go-slices-usage-and-internals

calling `len` is precisely reading a length field in an struct.